### PR TITLE
[GUILD] - Convert `Radio` to TS

### PR DESCRIPTION
**Pull request by eniskraasniqi1**
### Description

- This PR converts Radio component to typescript

### Manual check

- Check storybook, it should be the same as the live one.

### CODE REVIEW

**Review by JorenSaeyTL — APPROVED** [review]

File: src/components/radio/RadioGroup.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import omit from 'lodash.omit';
+import { GenericComponent } from '../../@types/types';
+import isReactElement from '../utils/isReactElement';
+import { BoxProps } from '../box/Box';
+
+export interface RadioGroupProps extends Omit<BoxProps, 'children' | 'className'> {
+  children?: ReactNode;
+  className?: string;
+  disabled?: boolean;
+  onChange?: (value: string | boolean, event?: React.SyntheticEvent<Element, Event>) => void;
```
Comment (qubis741) [inline src/components/radio/RadioGroup.tsx:14]: I think `ChangeEvent` is better here

File: src/components/radio/RadioGroup.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  children,
+  className = '',
+  disabled = false,
+  onChange,
+  value,
+  ...others
+}) => {
+  const rest = omit(others, ['onChange']);
+
+  const handleChange = (value: string | boolean, event: React.SyntheticEvent<Element, Event>) => {
```
Comment (qubis741) [inline src/components/radio/RadioGroup.tsx:28]: I think `ChangeEvent` is better here

File: src/components/radio/RadioGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        const isRadioButtonComponent = isComponentOfType(RadioButton, child);
+
+        if (!isRadioButtonComponent) {
+          return child;
+        }
+        if (isRadioButtonComponent && isReactElement(child)) {
+          return React.cloneElement(child, {
+            checked: child.props.value === value,
+            disabled: disabled || child.props.disabled,
+            onChange: (event: React.SyntheticEvent<Element, Event>) => handleChange(child.props.value, event),
```
Comment (qubis741) [inline src/components/radio/RadioGroup.tsx:46]: I think `ChangeEvent` is better here

File: src/components/radio/RadioButton.tsx [hunk 4]
```diff
@@ -26,7 +25,6 @@ const RadioButton: GenericComponent<RadioButtonProps> = ({
   children,
   className,
   disabled = false,
-  name,
```
Comment (qubis741) [inline src/components/radio/RadioButton.tsx]: its actually used, but its hidden in old spread, I will fix this

**Review by qubis741 — DISMISSED** [review]

**Comment by qubis741** [comment]
Manually checked :white_check_mark:

**Review by lowiebenoot — APPROVED** [review]

**Review by qubis741 — APPROVED** [review]

**Review by qubis741 — APPROVED** [review]

**Review by lowiebenoot — APPROVED** [review]

**Pull request merged**